jplag/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -1,320 +1,265 @@
-        this.oldSubmissionDirectories = oldSubmissionDirectories;
-        this.languageIdentifier = languageIdentifier;
+    private static final Logger logger = LoggerFactory.getLogger(JPlag.class);
+
+    public JPlagOptions(Language language, List<String> submissionDirectories, List<String> oldSubmissionDirectories) {
+        this(language, //
+                DEFAULT_COMPARISON_MODE, //
+                false, //
+                null, //
+                DEFAULT_SIMILARITY_THRESHOLD, //
+                DEFAULT_SHOWN_COMPARISONS, //
+                DEFAULT_SIMILARITY_METRIC, //
+                null, //
+                null, //
+                submissionDirectories, //
+                oldSubmissionDirectories, //
+                null, //
+                null, //
+                null, //
+                new ClusteringOptions.Builder().build() //
+        );
+    }
+
+    public JPlagOptions(Language language, ComparisonMode comparisonMode, boolean debugParser, List<String> fileSuffixes, float similarityThreshold,
+            Integer maximumNumberOfComparisons, SimilarityMetric similarityMetric, Integer minimumTokenMatch, String exclusionFileName,
+            List<String> submissionDirectories, List<String> oldSubmissionDirectories, String baseCodeSubmissionName, String subdirectoryName,
+            Verbosity verbosity, ClusteringOptions clusteringOptions) {
+        this.language = language;
+        this.comparisonMode = comparisonMode;
+        this.debugParser = debugParser;
+        this.fileSuffixes = fileSuffixes == null ? null : fileSuffixes.stream().toList();
+        this.similarityThreshold = fixSimilarityThreshold(similarityThreshold);
+        this.maximumNumberOfComparisons = fixMaximumNumberOfComparisons(maximumNumberOfComparisons);
+        this.similarityMetric = similarityMetric;
+        this.minimumTokenMatch = minimumTokenMatch;
+        this.exclusionFileName = exclusionFileName;
+        this.submissionDirectories = submissionDirectories == null ? null : submissionDirectories.stream().toList();
+        this.oldSubmissionDirectories = oldSubmissionDirectories == null ? null : oldSubmissionDirectories.stream().toList();
+        this.baseCodeSubmissionName = baseCodeSubmissionName;
+        this.subdirectoryName = subdirectoryName;
+        this.verbosity = verbosity;
+        this.clusteringOptions = clusteringOptions;
     }
 
-    public Optional<String> getBaseCodeSubmissionName() {
-        return Optional.ofNullable(baseCodeSubmissionName);
+    public JPlagOptions withLanguageOption(Language language) {
+        return new JPlagOptions(language, comparisonMode, debugParser, fileSuffixes, similarityThreshold, maximumNumberOfComparisons,
+                similarityMetric, minimumTokenMatch, exclusionFileName, submissionDirectories, oldSubmissionDirectories, baseCodeSubmissionName,
+                subdirectoryName, verbosity, clusteringOptions);
```

Comment:
Do we need all of these factory methods? E.g. I would say that the language or also the submission details should be set only when initially instantiating the options.

Reply:
I would like to have factories for all options. Otherwise, we have to discuss use cases for any option.